src/Graphics/UI/FLTK/LowLevel/Ask.chs

```diff
@@ -36,13 +36,23 @@ flBeep :: Maybe BeepType -> IO ()
 flBeep Nothing = flBeep'
 flBeep (Just bt) = flBeepType' (fromIntegral (fromEnum bt))
 
-{# fun flc_input as flInput' { unsafeToCString `T.Text' } -> `()' #}
-flInput :: T.Text -> IO ()
-flInput = flInput'
-
-{# fun flc_password as flPassword' { unsafeToCString `T.Text' } -> `()' #}
-flPassword :: T.Text -> IO ()
-flPassword = flPassword'
+{# fun flc_input as flInput' { unsafeToCString `T.Text' } -> `Maybe T.Text' unsafeFromMaybeCString #}
+flInput :: T.Text -> IO (Maybe T.Text)
+flInput msg = do
+  r <- flInput' msg
+
+  -- force the result, otherwise multiple calls to 'flInput' may appear to have
+  -- the same result even if the user typed different things
+  r `seq` return r
```

Comment:
I didn't see similar forcing code near the other `unsafeFromCString` calls, are there perhaps bugs at all of those locations?

Reply:
You're right! I'm going to remove all `unsafePerformIO` calls from the codebase over the next day or two. They were a mistake.